mobile/src/main/java/ch/epfl/sdp/mobile/ui/prepare_game/OpponentList.kt

```diff
@@ -0,0 +1,117 @@
+package ch.epfl.sdp.mobile.ui.prepare_game
+
+import androidx.compose.foundation.ExperimentalFoundationApi
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.LazyListState
+import androidx.compose.foundation.lazy.items
+import androidx.compose.foundation.lazy.rememberLazyListState
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.material.*
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.compositeOver
+import androidx.compose.ui.unit.dp
+import ch.epfl.sdp.mobile.state.toColor
+import ch.epfl.sdp.mobile.ui.social.Person
+
+/**
+ * Composable that represents a list of potential opponents, with one potentially selected
+ * @param state the state of the [PrepareGameScreen]
+ * @param modifier [Modifier] for this composable
+ * @param lazyListState the state of the lazyList displaying opponents
+ * @param key a function which uniquely identifies the list items.
+ * @param P the type of the [Person].
+ */
+@OptIn(ExperimentalFoundationApi::class)
+@Composable
+fun <P : Person> OpponentList(
+    opponents: List<P>,
+    selectedOpponent: P?,
+    onOpponentClick: (P) -> Unit,
+    modifier: Modifier = Modifier,
+    lazyListState: LazyListState = rememberLazyListState(),
+    key: ((P) -> Any)? = null,
+) {
+  LazyColumn(
+      verticalArrangement = Arrangement.spacedBy(8.dp),
+      horizontalAlignment = Alignment.CenterHorizontally,
+      modifier = modifier,
+      contentPadding = PaddingValues(8.dp),
+      state = lazyListState,
+  ) {
+    items(
+        items = opponents,
+        key = key,
+        itemContent = { item ->
+          Opponent(
+              item,
+              onClick = { onOpponentClick(item) },
+              modifier = Modifier.animateItemPlacement(),
+              selected = item == selectedOpponent,
+          )
+        },
+    )
+  }
+}
+
+/**
+ * Composable that represents an opponent in [OpponentList]
+ * @param person Person representing the opponent
+ * @param onClick button action
+ * @param modifier [Modifier] for this composable
+ * @param selected indicates if the opponent is selected
+ * @param P the type of the [Person].
+ */
+@OptIn(ExperimentalMaterialApi::class)
+@Composable
+private fun <P : Person> Opponent(
+    person: P,
+    onClick: () -> Unit,
+    modifier: Modifier = Modifier,
+    selected: Boolean = false,
+) {
+  Button(
+      modifier = modifier.fillMaxWidth(),
```

Comment:
It should be the caller's responsibility to choose whether the `Opponent` should fill the whole screen width probably and apply the `Modifier`.

```suggestion
      modifier = modifier,
```

Reply:
Fixed in [fd0fdcb](https://github.com/epfl-SDP/android/pull/200/commits/fd0fdcbac984e142120bda85975f53f8d93f6489)